dm/syncer/syncer_test.go

```diff
@@ -956,6 +956,7 @@ func (s *testSyncerSuite) TestRun(c *C) {
 	}
 
 	cancel()
+	<-resultCh // wait for the process to finish
```

Comment:
Just a question, can you help to explain some reason about why syncer does not finish when we called `cancel()`.

Reply:
there are  lots of backgraound [jobs ](https://github.com/pingcap/tiflow/blob/4bfa080ecce6c57a670af290a764e2654e9361bb/dm/syncer/syncer.go#L1541)in syncer.Run and they may block main thread exit for little time